src/core/server/saved_objects/migrationsv2/actions/index.ts

```diff
@@ -319,6 +319,20 @@ const waitForTask = (
         description: body.task.description,
       });
     })
+    .catch((e) => {
+      if (
+        e.body?.error?.type === 'timeout_exception' ||
+        e.body?.error?.type === 'receive_timeout_transport_exception'
+      ) {
+        return Either.left({
+          type: 'retryable_es_client_error' as const,
```

Comment:
shouldn't we place this logic in `catchRetryableEsClientErrors` then?

Reply:
I started there, but then thought, this isn't a generic retryable error like a 503 or a socket timeout, these error codes are specific to the `_tasks` API, so I thought it makes sense to include this logic here. WDYT?

Reply:
Oh I see it now, the error type name `retryable_es_client_error` is tied to errors originating from the ES client. I think we should rather have a specific error type so that the signature of `waitForTask` shows that callers should expect this "your task didn't complete within the timeout" response.

Then the model could handle this and explicitly call `delayRetryState`. We can change the signature of `delayRetryState` to accept a maxRetryAttempts parameter. That way we can maybe retry `waitForTask` steps forever (or maybe just a really large limit equivalent to 24h).

'retryable_es_client_error' could be any generic problem like a cluster behind a proxy being offline, in such a case I don't think we want to keep retrying forever, we'd rather fail relatively fast so there's an error log users can investigate.

But if we're waiting for a task, and we can successfully get the task status from ES and ES is just saying that it's still busy, then everything is working, even if it's slow, so we can keep on waiting for the task to complete even if it takes hours.

Reply:
we also need to add an integration test for the `cloneIndex` operation if `waitForIndexStatusYellow` hits the 30s timeout